presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -546,4 +561,58 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
                 table.getTableName(),
                 oldStats -> updatePartitionStatistics(oldStats, tableStatistics));
     }
+
+    @Override
+    public void registerTable(ConnectorSession clientSession, SchemaTableName schemaTableName, Path metadataLocation)
+    {
+        String tableLocation = metadataLocation.getName();
+        HdfsContext hdfsContext = new HdfsContext(
+                clientSession,
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                tableLocation,
+                true);
+
+        InputFile inputFile = new HdfsInputFile(metadataLocation, hdfsEnvironment, hdfsContext);
+        TableMetadata tableMetadata;
+        try {
+            tableMetadata = TableMetadataParser.read(new HdfsFileIO(hdfsEnvironment, hdfsContext), inputFile);
+        }
+        catch (RuntimeException e) {
```

Comment:
Just catch Exception
```suggestion
        catch (Exception e) {
```